rust-symcrypt/src/hash.rs

```diff
@@ -195,13 +207,28 @@ pub struct Md5State(Pin<Box<symcrypt_sys::SYMCRYPT_MD5_STATE>>);
 #[cfg(feature = "md5")]
 impl Md5State {
     pub fn new() -> Self {
-        let mut instance = Md5State(Box::pin(symcrypt_sys::SYMCRYPT_MD5_STATE::default()));
+        let mut instance = Md5State(Box::pin(Md5InnerState { inner: symcrypt_sys::SYMCRYPT_MD5_STATE::default(), _pinned: PhantomPinned }));
         unsafe {
             // SAFETY: FFI calls
-            symcrypt_sys::SymCryptMd5Init(&mut *instance.0);
+            symcrypt_sys::SymCryptMd5Init(instance.get_inner_mut());
         }
         instance
     }
+
+    /// Get a mutable pointer to the inner SymCrypt state
+    ///
+    /// This is primarily meant to be used while making calls to the underlying SymCrypt APIs.
+    fn get_inner_mut(&mut self) -> *mut symcrypt_sys::SYMCRYPT_MD5_STATE {
+        unsafe { 
+            // This function returns pointer to pinned data, which means callers must not use the pointer to move the data out of its location.
```

Comment:
It'd be good to state the part about what callers need to do in the doc comment so that the callers can see it too :)

Reply:
This applies to the similar methods further down too.

Reply:
I'll update it to be more clear with doc comments as well